lib/Sema/CSSimplify.cpp

```diff
@@ -2175,6 +2175,66 @@ bool ConstraintSystem::repairFailures(
     }
     break;
   }
+  case ConstraintLocator::ApplyArgToParam: {
+    auto loc = getConstraintLocator(locator);
+    if (repairByInsertingExplicitCall(lhs, rhs))
+      return true;
+
+    if (lhs->getOptionalObjectType() && !rhs->getOptionalObjectType()) {
+      conversionsOrFixes.push_back(
+          ForceOptional::create(*this, lhs, lhs->getOptionalObjectType(),
+                                loc));
+    }
+
+    Expr *argAnchor = nullptr;
+    if (auto *AE = dyn_cast<ApplyExpr>(loc->getAnchor())) {
```

Comment:
Easier way to do this is to use - `simplifyLocatorToAnchor` - for apply argument path it would return argument you are looking for :)

Reply:
Ahh, glad to know that's a thing!